desc/coils.py

```diff
@@ -1665,7 +1670,7 @@ def __init__(self, *coils, name="", check_intersection=True):
     @property
     def num_coils(self):
         """int: Number of coils."""
-        return sum([c.num_coils if hasattr(c, "num_coils") else 1 for c in self])
+        return sum([c.num_coils for c in self])
```

Comment:
Does this work for a nested coilset?

Reply:
I think it should, because num_coils will just get called recursively until the self is a single coilset